hardhat.config.ts

```diff
@@ -0,0 +1,41 @@
+import { HardhatUserConfig } from "hardhat/config";
+import "@nomicfoundation/hardhat-toolbox";
+
+const config: HardhatUserConfig = {
+    solidity: {
+        version: '0.5.17', // todo update solidity to 0.8.xx
+        settings: {
+            optimizer: {
+                enabled: true,
+                runs: 200,
+            },
+        },
+    },
+    networks: {
+        hardhat: {
+            allowUnlimitedContractSize: true,
+        },
+        localhost: {
+            url: 'http://127.0.0.1:8545',
+        },
+        mainnet: {
+            url: 'https://rpc.sonic.soniclabs.com',
+            chainId: 146,
+            hardfork: "cancun",
+        },
+        testnet: {
+            url: 'https://rpc.blaze.soniclabs.com',
+            chainId: 57054,
+            hardfork: "cancun",
+        },
+    },
```

Comment:
custom chain for sonicscan:
https://github.com/Fantom-foundation/Bridge/blob/jk/mainnet-deployment2/hardhat.config.ts